exporters/otlp/otlplog/otlploggrpc/internal/transform/log_test.go

```diff
@@ -182,76 +219,97 @@ var (
 			ObservedTimeUnixNano: uint64(obs.UnixNano()),
-			SeverityNumber:       pbSevB,
-			SeverityText:         "B",
-			Body:                 pbBodyB,
-			Attributes:           []*cpb.KeyValue{pbAlice},
-			Flags:                uint32(flagsB),
-			TraceId:              traceIDB,
-			SpanId:               spanIDB,
+			SeverityNumber:       pbSevD,
+			SeverityText:         "D",
+			Body:                 pbBodyD,
+			Attributes:           []*cpb.KeyValue{pbTom},
+			Flags:                uint32(flagsD),
+			TraceId:              traceIDD,
+			SpanId:               spanIDD,
 		},
 		{
 			TimeUnixNano:         uint64(ts.UnixNano()),
 			ObservedTimeUnixNano: uint64(obs.UnixNano()),
-			SeverityNumber:       pbSevB,
-			SeverityText:         "B",
-			Body:                 pbBodyB,
-			Attributes:           []*cpb.KeyValue{pbBob},
-			Flags:                uint32(flagsB),
-			TraceId:              traceIDB,
-			SpanId:               spanIDB,
+			SeverityNumber:       pbSevD,
+			SeverityText:         "D",
+			Body:                 pbBodyD,
+			Attributes:           []*cpb.KeyValue{pbJerry},
+			Flags:                uint32(flagsD),
+			TraceId:              traceIDD,
+			SpanId:               spanIDD,
 		},
 		{
 			TimeUnixNano:         0,
 			ObservedTimeUnixNano: uint64(obs.UnixNano()),
-			SeverityNumber:       pbSevB,
-			SeverityText:         "B",
-			Body:                 pbBodyB,
-			Attributes:           []*cpb.KeyValue{pbBob},
-			Flags:                uint32(flagsB),
-			TraceId:              traceIDB,
-			SpanId:               spanIDB,
+			SeverityNumber:       pbSevD,
+			SeverityText:         "D",
+			Body:                 pbBodyD,
+			Attributes:           []*cpb.KeyValue{pbJerry},
+			Flags:                uint32(flagsD),
+			TraceId:              traceIDD,
+			SpanId:               spanIDD,
 		},
 	}
 
-	pbScopeLogs = &lpb.ScopeLogs{
-		Scope:      pbScope,
-		SchemaUrl:  semconv.SchemaURL,
-		LogRecords: pbLogRecords,
+	pbScopeLogsList = []*lpb.ScopeLogs{
+		{
+			Scope:      pbScope,
+			SchemaUrl:  semconv.SchemaURL,
+			LogRecords: pbLogRecords,
+		},
+		{
+			Scope:      pbScope2,
+			SchemaUrl:  semconv.SchemaURL,
+			LogRecords: pbLogRecords,
+		},
 	}
 
-	pbResourceLogs = &lpb.ResourceLogs{
-		Resource:  pbRes,
-		SchemaUrl: semconv.SchemaURL,
-		ScopeLogs: []*lpb.ScopeLogs{pbScopeLogs},
+	pbResourceLogsList = []*lpb.ResourceLogs{
+		{
+			Resource:  pbRes,
+			SchemaUrl: semconv.SchemaURL,
+			ScopeLogs: pbScopeLogsList,
+		},
+		{
+			Resource:  pbRes2,
+			SchemaUrl: semconv.SchemaURL,
+			ScopeLogs: pbScopeLogsList,
+		},
 	}
 )
 
 func TestResourceLogs(t *testing.T) {
-	want := []*lpb.ResourceLogs{pbResourceLogs}
-	assert.Equal(t, want, ResourceLogs(records))
+	want := pbResourceLogsList
+	rLogs := ResourceLogs(records)
+	for _, got := range rLogs {
+		expected := want[1]
+		if got.Resource.Attributes[1].GetValue().GetStringValue() == "v0.1.1" {
+			expected = want[0]
+		}
+		assert.Equal(t, expected, got)
+	}
```

Comment:
```suggestion
	assert.Equal(t, want, ResourceLogs(records))
```

nit. This should be more clear.

Reply:
when we construct the array of ResourceLogs we internally create a map and that changes the order so when comparing it directly it definitely is more clear but will make this test flaky depending on the order.

Reply:
If you want to ignore order you can use https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

Reply:
@pellared thank you for the suggestion, I learned something new today :)
 I have made the changes using this function and pushed.